### PR TITLE
Add problem

### PR DESCRIPTION
**Pull request by HappyFeet07**
add freeshop problem

### CODE REVIEW

File: gamedata/deploy.ropsten.json [hunk 1]
```diff
@@ -2,5 +2,6 @@
   "0": "0x8b1b00be6b60739f8602f6cdda67a79f746555c0",
   "1": "0xe04d0f4fde42df86941d2b1c54bd22185f4219b0",
   "2": "0x16ab2bfe41acd8bc97a6e6b5570a93f701fd47b8",
+  "3": "0x184F9342692aC674b21CA3f0F802Cb9dE652ca46",
```
Comment (StillFantastic) [inline gamedata/deploy.ropsten.json]: Convert the address to lowercase

File: contracts/levels/FreeshopFactory.sol [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+import './base/Level.sol';
+import './Freeshop.sol';
+
+contract FreeshopFactory is Level {
+
+  function createInstance(address _player) public payable returns (address) {
+    _player;
+    Freeshop nowinstance = new Freeshop();
+    address(nowinstance).transfer(5 ether);
```
Comment (StillFantastic) [inline contracts/levels/FreeshopFactory.sol]: I think 1 ether is enough

File: contracts/levels/FreeshopFactory.sol [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    return nowinstance;
+  }
+
+  function validateInstance(address _instance, address _player) public returns (bool) {
+    _player;
+    Freeshop instance = Freeshop(_instance);
+    return instance.GetBalance() < 1 ether;
```
Comment (StillFantastic) [inline contracts/levels/FreeshopFactory.sol]: I think GetBalance() should equal to 0 is better

File: gamedata/gamedata.json [hunk 4]
```diff
@@ -44,6 +44,21 @@
...
+      "difficulty" : "3",
+      "description" : "freeshop.md",
+      "completedDescription" : "freeshop_complete.md",
+      "levelContract" : "Freeshop.sol",
+      "instanceContract": "FreeshopFactory.sol",
+      "revealCode" : true,
+      "deployParams": [],
+      "deployFunds": 0,
+      "deployId" : "3",
+      "instanceGas" : 300000,
```
Comment (StillFantastic) [inline gamedata/gamedata.json]: Should set a proper gaslimit

File: gamedata/gamedata.json [hunk 5]
```diff
@@ -44,6 +44,21 @@
...
+      "name" : "Freeshop",
+      "created" : "2019-09-26",
+      "difficulty" : "3",
+      "description" : "freeshop.md",
+      "completedDescription" : "freeshop_complete.md",
+      "levelContract" : "Freeshop.sol",
+      "instanceContract": "FreeshopFactory.sol",
+      "revealCode" : true,
+      "deployParams": [],
+      "deployFunds": 0,
```
Comment (StillFantastic) [inline gamedata/gamedata.json]: deployFunds is the amount of Ether to be sent while deploying

File: contracts/levels/Freeshop.sol [hunk 6]
```diff
@@ -0,0 +1,27 @@
+pragma solidity ^0.4.24;
+
+contract Freeshop {
+
+    uint256 public withdrawalLimit = 1 ether;
```
Comment (StillFantastic) [inline contracts/levels/Freeshop.sol]: maybe 0.1 Ether

File: contracts/levels/Freeshop.sol [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+        require(now >= lastWithdrawTime[msg.sender] + 1 weeks);
+        require(msg.sender.call.value(_weiToWithdraw)());
+        balances[msg.sender] -= _weiToWithdraw;
+        lastWithdrawTime[msg.sender] = now;
+    }
+    
+    function GetBalance() public view returns(uint){
+        return address(this).balance;
+    }
+ }
```
Comment (StillFantastic) [inline contracts/levels/Freeshop.sol]: Shoud have fallback function

**Comment by HappyFeet07** [comment]
@StillFantastic  review again please!!

File: contracts/levels/FreeshopFactory.sol [hunk 8]
```diff
@@ -16,7 +16,7 @@ contract FreeshopFactory is Level {
   function validateInstance(address _instance, address _player) public returns (bool) {
     _player;
     Freeshop instance = Freeshop(_instance);
-    return instance.GetBalance() < 0.1 ether;
+    return instance.GetBalance() = 0 ether;
```
Comment (StillFantastic) [inline contracts/levels/FreeshopFactory.sol]: ==

Comment (HappyFeet07) [reply contracts/levels/FreeshopFactory.sol]: ==

**Pull request merged**